src/App.tsx

```diff
@@ -42,54 +43,114 @@ export function getReorderedGoods(
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  render() {
+    const visibleGoods = getReorderedGoods(goodsFromServer, this.state);
+    const { sortType, isReversed } = this.state;
+
+    visibleGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
 
-      <ul>
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={
+              classNames(
+                'button',
+                'is-info',
+                { 'is-light': sortType !== SortType.ALPHABET },
+              )
+            }
+            onClick={() => {
+              this.setState({ sortType: SortType.ALPHABET });
+            }}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={
+              classNames(
+                'button',
+                'is-success',
+                { 'is-light': sortType !== SortType.LENGTH },
+              )
+            }
+            onClick={() => {
+              this.setState({ sortType: SortType.LENGTH });
+            }}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={
+              classNames(
+                'button',
+                'is-warning',
+                { 'is-light': !isReversed },
+              )
+            }
+            onClick={() => {
+              this.setState(state => ({
+                isReversed: !state.isReversed,
+              }));
+            }}
+          >
+            Reverse
+          </button>
+
+          {(isReversed || sortType !== SortType.NONE) && (
```

Comment:
It would be great to save all these booleans into some variable, so it's clear why you need them